File: backend/src/main/java/com/hentetjeneste/model/DailyInfo.java
package com.hentetjeneste.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "daily_info")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DailyInfo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private LocalDate date;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private InfoType type;
    
    @Column(nullable = false)
    private String title;
    
    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;
    
    private String targetGroup; // Optional: specific group
    
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();
    
    public enum InfoType {
        MENU, ACTIVITY, ANNOUNCEMENT
    }
}

File: src/src/types/dailyInfo.ts
export type DailyInfoType = 'menu' | 'activity' | 'announcement';

export interface DailyInfo {
  id: string;
  date: string;
  type: DailyInfoType;
  title: string;
  description: string;
  target_group: string | null;
  created_by: string | null;
  created_at: string;
  updated_at: string;
}

export interface DailyInfoInput {
  date: string;
  type: DailyInfoType;
  title: string;
  description: string;
  target_group?: string | null;
}
